spring-integration-core/src/test/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandlerTests.java

```diff
@@ -186,20 +187,32 @@ public void testReaperReapsAnEmptyGroupAfterConfiguredDelay() throws Exception {
 		});
 		handler.setReleaseStrategy(group -> group.size() == 1);
 
-		handler.setMinimumTimeoutForEmptyGroups(1000);
-
-		Message<String>	message = MessageBuilder.withPayload("foo")
+		Message<String> message = MessageBuilder.withPayload("foo")
 				.setCorrelationId("bar")
 				.build();
 		handler.handleMessage(message);
 
+		handler.setMinimumTimeoutForEmptyGroups(100);
+
 		assertEquals(1, outputMessages.size());
 
 		assertEquals(1, TestUtils.getPropertyValue(handler, "messageStore.groupIdToMessageGroup", Map.class).size());
 		groupStore.expireMessageGroups(0);
 		assertEquals(1, TestUtils.getPropertyValue(handler, "messageStore.groupIdToMessageGroup", Map.class).size());
-		Thread.sleep(1010);
-		groupStore.expireMessageGroups(0);
+
+		int n = 0;
+
+		while (n++ < 100) {
```

Comment:
200 or s/50/100/ below - we usually use 10 seconds sometimes 5 is not enough for Bamboo 😄